pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -432,14 +416,19 @@ func newQueryMiddlewares(
 		prom2CompatMiddleware,
 	)
 
+	rangeQuerySpinOffMiddlewares := append([]MetricsQueryMiddleware{
+		splitAndCacheMiddleware,
+	}, cfg.ExtraRangeQueryMiddlewares...)
```

Comment:
my thinking was that the split and cache will only be a detour into the split-and-cache if we have to do it. After the split-and-cache the request will still go through the rest of the middlewares all instant queries go through (cardinality, sharding, pruning retry, experimental features)

why do we need the extra middlewares here again?